diff --git a/video_embed_field.handlers.inc b/video_embed_field.handlers.inc
index 1962008..b9fe593 100644
--- a/video_embed_field.handlers.inc
+++ b/video_embed_field.handlers.inc
@@ -195,7 +195,7 @@ function video_embed_field_handle_youtube($url, $settings) {
 
   // Construct the embed code.
   $settings['wmode'] = 'opaque';
-  $settings_str = urlencode(_video_embed_code_get_settings_str($settings));
+  $settings_str = _video_embed_code_get_settings_str($settings);
 
   $output['#markup'] = '<iframe class="' . check_plain($class) . '" width="' . check_plain($settings['width']) . '" height="' . check_plain($settings['height']) . '" src="//www.youtube.com/embed/' . $id . '?' . $settings_str . '" frameborder="0" allowfullscreen></iframe>';
 
diff --git a/video_embed_field.module b/video_embed_field.module
index 8826ead..e4f4282 100644
--- a/video_embed_field.module
+++ b/video_embed_field.module
@@ -617,7 +617,7 @@ function _video_embed_code_get_settings_str($settings = array()) {
       $values[] = $name;
     }
     else {
-      $values[] = $name . '=' . $value;
+      $values[] = $name . '=' . urlencode($value);
     }
   }
 
